Content.Shared/Ninja/Systems/SharedNinjaSuitSystem.cs

```diff
@@ -0,0 +1,174 @@
+using Content.Shared.Stealth;
+using Content.Shared.Stealth.Components;
+using Content.Shared.Timing;
+using Robust.Shared.Audio;
+using Robust.Shared.Network;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Serialization;
+
+namespace Content.Shared.Ninja.Systems;
+
+public abstract class SharedNinjaSuitSystem : EntitySystem
+{
+    [Dependency] private readonly SharedAudioSystem _audio = default!;
+    [Dependency] private readonly SharedNinjaGlovesSystem _gloves = default!;
+    [Dependency] protected readonly SharedSpaceNinjaSystem _ninja = default!;
+    [Dependency] private readonly SharedStealthSystem _stealth = default!;
+    [Dependency] protected readonly UseDelaySystem _useDelay = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<NinjaSuitComponent, GotEquippedEvent>(OnEquipped);
+        SubscribeLocalEvent<NinjaSuitComponent, GetItemActionsEvent>(OnGetItemActions);
+        SubscribeLocalEvent<NinjaSuitComponent, GotUnequippedEvent>(OnUnequipped);
+
+        SubscribeNetworkEvent<SetCloakedMessage>(OnSetCloakedMessage);
+    }
+
+    /// <summary>
+    /// Call the shared and serverside code for when a ninja equips the suit.
+    /// </summary>
+    private void OnEquipped(EntityUid uid, NinjaSuitComponent comp, GotEquippedEvent args)
+    {
+        var user = args.Equipee;
+        if (!TryComp<SpaceNinjaComponent>(user, out var ninja))
+            return;
+
+        NinjaEquippedSuit(uid, comp, user, ninja);
+    }
+
+    /// <summary>
+    /// Add all the actions when a suit is equipped.
+    /// Since the event doesn't pass user this can't check if it's a ninja early and not add actions.
+    /// </summary>
+    private void OnGetItemActions(EntityUid uid, NinjaSuitComponent comp, GetItemActionsEvent args)
+    {
+        args.Actions.Add(comp.TogglePhaseCloakAction);
+        args.Actions.Add(comp.RecallKatanaAction);
+        args.Actions.Add(comp.CreateThrowingStarAction);
+        args.Actions.Add(comp.KatanaDashAction);
+        args.Actions.Add(comp.EmpAction);
+    }
+
+    /// <summary>
+    /// Call the shared and serverside code for when anyone unequips a suit.
+    /// </summary>
+    private void OnUnequipped(EntityUid uid, NinjaSuitComponent comp, GotUnequippedEvent args)
+    {
+        UserUnequippedSuit(uid, comp, args.Equipee);
+    }
+
+    /// <summary>
+    /// Called when a suit is equipped by a space ninja.
+    /// In the future it might be changed to an explicit activation toggle/verb like gloves are.
+    /// </summary>
+    protected virtual void NinjaEquippedSuit(EntityUid uid, NinjaSuitComponent comp, EntityUid user, SpaceNinjaComponent ninja)
+    {
+        // mark the user as wearing this suit, used when being attacked among other things
+        _ninja.AssignSuit(ninja, uid);
+
+        // initialize phase cloak
+        EnsureComp<StealthComponent>(user);
+        SetCloaked(user, comp.Cloaked);
```

Comment:
The suit itself should just relay to the wearer instead of adding stealth to the user as this falls over as soon as multiple things try to touch stealth.

Reply:
is that possible, wouldnt that just make the suit invisible

Reply:
in its own pr so resolving